### PR TITLE
Issue/592/ccl v3

### PR DESCRIPTION
**Pull request by hsinfan1996**
- [x] Update _ccl_supported_versions, README and INSTALL accordingly after merging main
- [x] Remove warning filter in the example notebooks (CCL removed problematic warnings in v2.7.1.dev10, which is the minimum supported version proposed by this PR)
- [x] Update `tests/data/numcosmo/generate_einasto_hernquist_benchmarks_fromNC.ipynb`
- [x] Update `demo_mock_ensemble_realistic.ipynb`
- [x] Add option to use _projected_quad for the CCL backend (both OO and functional interfaces)
- [x] Add tests for CCL virial massdef.
- [x] Drop CT virial massdef.
- [x] Show how to enable `projected_quad`
- [x] Partially address #491

### CODE REVIEW

File: clmm/theory/ccl_v3.py
Comment (hsinfan1996) [inline clmm/theory/ccl_v3.py:128]: This line is different from that of v2. In v3 `self.mdef` is not needed for `_get_alpha`.

**Comment by hsinfan1996** [comment]
If we are not supporting both v2 and v3 at the same time, I will delete this PR as this is not necessary. However, we will need to update README to specify the maximum or minimum version of CCL supported.

**Comment by m-aguena** [comment]
I don't think it will be feasible to maintain support for both versions. We should just keep track which releases are currently supported.

**Comment by hsinfan1996** [comment]
- [ ] Update _ccl_supported_versions, README and INSTALL accordingly after merging main

**Comment by hsinfan1996** [comment]
Don't know if anyone is using CCL v2.8, but this PR is for them and is ready. We can still wait for the official release of CCL v3.

**Comment by m-aguena** [comment]
@hsinfan1996 do you know if there is an ETA for CCL v3?

**Comment by hsinfan1996** [comment]
> @hsinfan1996 do you know if there is an ETA for CCL v3?

I think there is no exact date for the release of CCL v3. What I know is that before the official release of v3, there are still some tweaks needed on documentation and example notebooks, which can be found here https://github.com/LSSTDESC/CCL/milestone/5.

**Comment by hsinfan1996** [comment]
CCL v3 will be tagged on Monday.

**Comment by m-aguena** [comment]
@hsinfan1996 when you were talking about removing `virial` mass from cluster_toolkit, you meant line 88 of `clmm/theory/cluster_toolkit.py`?

**Comment by hsinfan1996** [comment]
> @hsinfan1996 when you were talking about removing `virial` mass from cluster_toolkit, you meant line 88 of `clmm/theory/cluster_toolkit.py`?

Yes. I added that line so it was consistent with CCL backend at that time.

**Comment by m-aguena** [comment]
> > @hsinfan1996 when you were talking about removing `virial` mass from cluster_toolkit, you meant line 88 of `clmm/theory/cluster_toolkit.py`?
> 
> Yes. I added that line so it was consistent with CCL backend at that time.

I see, so it can be removed now. Just remember to also update the docstring of `CTCLMModeling`

File: clmm/theory/parent_class.py
Comment (m-aguena) [inline clmm/theory/parent_class.py:451]: can you change `projected_quad` to something like `use_projected_quad`?
Also do this for `_set_projected_quad` functions

**Review by m-aguena — CHANGES_REQUESTED** [review]
I think it would be a good idea to add `use_projected_quad` in the `theory.func_layer` functions. Also add examples in notebooks

**Review by m-aguena — CHANGES_REQUESTED** [review]
There are a few notebooks that are failing due CCL v3.0 API changes, can you update them?
```
Paper_v1.0/mcmc.ipynb
Paper_v1.0/validation_tests.ipynb
Paper_v1.0/mass_bias_from_models.ipynb
other_compare_NFW_critical_massdef.ipynb
```
Thanks

**Comment by hsinfan1996** [comment]
> There are a few notebooks that are failing due CCL v3.0 API changes, can you update them?
> 
> ```
> 
> Paper_v1.0/mcmc.ipynb
> 
> Paper_v1.0/validation_tests.ipynb
> 
> Paper_v1.0/mass_bias_from_models.ipynb
> 
> other_compare_NFW_critical_massdef.ipynb
> 
> ```
> 
> Thanks 

Yes, will do. I thought they should be in the state as when the paper was published for archival purposes.

**Comment by m-aguena** [comment]
> > There are a few notebooks that are failing due CCL v3.0 API changes, can you update them?
> > ```
> > 
> > Paper_v1.0/mcmc.ipynb
> > 
> > Paper_v1.0/validation_tests.ipynb
> > 
> > Paper_v1.0/mass_bias_from_models.ipynb
> > 
> > other_compare_NFW_critical_massdef.ipynb
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Thanks
> 
> Yes, will do. I thought they should be in the state as when the paper was published for archival purposes.

That version will still be able to be accessed by going to CLMM v1.0

**Comment by hsinfan1996** [comment]
Don't know how to update "examples/Paper_v1.0/mcmc.ipynb". Maybe leave it to issue #606.

**Comment by m-aguena** [comment]
> Don't know how to update "examples/Paper_v1.0/mcmc.ipynb". Maybe leave it to issue #606 since it is related to NumCosmo.

You are right, this is related to NumCosmo

**Review by m-aguena — APPROVED** [review]

**Comment by hsinfan1996** [comment]
Forcing cmake=3.27.4 makes checks run again, but makes solving environment very slow. Want to wait a little longer to see if everything fixes itself again.

**Comment by hsinfan1996** [comment]
Fixed GH Actions by adding `--override-channels` when installing cmake.

**Comment by hsinfan1996** [comment]
Not sure why, maybe related to the release of cmake 3.27.5, conda cannot solve the environment consistently and get package from conda-forge if `--override-channels` is not issued.

**Comment by hsinfan1996** [comment]
Only cmake from conda-forge works regardless of the versions.

**Comment by hsinfan1996** [comment]
On a side note, numpy 1.26 is compatible.

**Comment by m-aguena** [comment]
> Not sure why, maybe related to the release of cmake 3.27.5, conda cannot solve the environment consistently and get all packages from conda-forge if `--override-channels` is not issued. On my computer, issuing `conda install -c conda-forge cmake swig` can get different results. Most notably, `swig` may be installed from conda or conda-forge every time I issue the same command.

This will do for now. We can keep an issue to see if it will be solved by a newer version of conda

**Pull request merged**
